plugins/modules/iam_user.py

```diff
@@ -273,15 +281,31 @@ def create_or_update_login_profile(connection, module):
     return True, retval
 
 
-def delete_login_profile(connection, module):
-
-    user_params = dict()
-    user_params['UserName'] = module.params.get('name')
+def delete_login_profile(connection, module, user_name, wait):
+    '''
+    Deletes a users login profile.
+        Parameters:
+            connection: IAM client
+            module: AWSModule
+            user_name (str): Username of user
+            wait (bool): Whether or not the module waits for full deletion to reflect on AWS
+        Returns:
+            (bool): True if login profile deleted, False if no login profile found to delete
+    '''
+    # User does not have login profile - nothing to delete
+    if not user_has_login_profile(connection, module, user_name):
+        return False
 
-    try:
-        connection.delete_login_profile(**user_params)
-    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
-        module.fail_json_aws(e, msg="Unable to delete user login profile")
+    if not module.check_mode:
+        try:
+            connection.delete_login_profile(UserName=user_name, aws_retry=True)
+            if wait:
+                # Wait for login profile to be completely removed
+                connection.delete_login_profile(UserName=user_name, aws_retry=True)
+        except is_boto3_error_code('NoSuchEntity'):
```

Comment:
Could it be this except here is redundant? The case when there is no profile seems be handled by https://github.com/ansible-collections/community.aws/pull/1059/files#diff-ed1dcdfbd9adda436c1642a9623c2b9cdabe8c6979c93c39aa03639db33f348eR296-R297, is it?

Reply:
I figured it wouldn't hurt to be safe, but yes it probably won't ever be called unless some weird timing issue occurs.

Reply:
@alinabuzachis its needed here for the "wait"

Reply:
You can create a custom waiter for a delete action using exception criteria, there's an example here: https://github.com/ansible-collections/amazon.aws/blob/main/plugins/module_utils/waiters.py#L111-L132

Reply:
marking as resolved as no waiter was needed